waspc/packages/deploy/src/providers/fly/deploy/deploy.ts

```diff
@@ -0,0 +1,119 @@
+import { exit } from 'process';
+import { $, cd, fs } from 'zx';
+import {
+	cdToClientBuildDir,
+	cdToServerBuildDir,
+	displayWaspRocketImage,
+	makeIdempotent,
+	getCommandHelp,
+	waspSays,
+} from '../helpers/helpers.js';
+import {
+	clientTomlExistsInProject,
+	copyLocalClientTomlToProject,
+	copyLocalServerTomlToProject,
+	copyProjectClientTomlLocally,
+	copyProjectServerTomlLocally,
+	getInferredBasenameFromClientToml,
+	getInferredBasenameFromServerToml,
+	getTomlFilePaths,
+	serverTomlExistsInProject,
+} from '../helpers/tomlFileHelpers.js';
+import { DeployOptions } from './DeployOptions.js';
+import { createDeploymentInfo, DeploymentInfo } from '../DeploymentInfo.js';
+import { flySetupCommand } from '../index.js';
+import { secretExists } from '../helpers/flyctlHelpers.js';
+
+export async function deploy(options: DeployOptions): Promise<void> {
+	waspSays('Deploying your Wasp app to Fly.io!');
+
+	const buildWasp = makeIdempotent(async () => {
+		if (options.skipBuild) {
+			return;
+		}
+
+		waspSays('Building your Wasp app...');
+		cd(options.waspProjectDir);
+		await $`${options.waspExe} build`;
+	});
+
+	const tomlFilePaths = getTomlFilePaths(options);
+
+	// NOTE: Below, it would be nice if we could store the client, server, and DB names somewhere.
+	// For now we just rely on the suffix naming convention and infer from toml files.
+	if (!serverTomlExistsInProject(tomlFilePaths)) {
```

Comment:
Should we support depoying only server or only client?

This assumes that "if server toml is missing, setup wasn't run" but it's okay to deploy client next. I have a feeling that our deployment should be atomic - it doesn't make sense to deploy only one part.

Maybe we should adjust the logic here: if either TOML is missing, abort and assume the setup wasn't run.

cc: @sodic @Martinsos to chip in as well

Reply:
If we want to allow users to deploy only one or the other, we should provide that option via CLI arguments.

Reply:
That is a good question. It was mainly there just in case one part failed, you could re-run the command and get the second part to work. So trying to make things like `setup` be idempotent, but still recoverable. I think making it atomic and checking for both files probably does make the most sense though.

But adding a `--skip-server` and `--skip-client` is interesting. Just not sure if we want to give that much flexibility just yet. 🤔 Let's see what others think, but I am fine making atomic and updating all checks for both vs individual.

Reply:
> That is a good question. It was mainly there just in case one part failed, you could re-run the command and get the second part to work.

After my `launch` command failed due to missing secrets (which automatically makes fly deploy the server), I had to rerun `deploy` to also deploy the client. 

Having the `--skip-server` would have helped, but I think it's no biggie that I redeployed server, only it took a bit longer.

Reply:
Ah yes, I agree that does sound useful. Ok, you convinced me, I think I'll just add the options to skip server/client for both `setup` and `deploy` commands :D I will leave `launch` doing both. Thanks!

Reply:
Sounds good to me!